HW2/P5/driver.py

```diff
@@ -75,8 +113,28 @@ def randcolor():
         if total_time > anim_step:
             animator.update(positions)
             print("{} simulation frames per second".format(frame_count / total_time))
+            total_frame_count += frame_count
+            total_total_time += total_time
             frame_count = 0
             total_time = 0
-            # SUBPROBLEM 3: sort objects by location.  Be sure to update the
-            # grid if objects' indices change!  Also be sure to sort the
-            # velocities with their object positions!
+            i += 1
+            if i == iterations:
+                print("Average: {} simulation frames per second".format(total_frame_count / total_total_time))
+                break
+            
+            # sort objects by location using a Morton ordering procedure            
+            grid_positions = (positions / grid_spacing).astype("int")
+            # map grid_positions + offset to the morton ordering function, then obtain the sorted indices
+            grid_indices = np.argsort(map(lambda x, y: cmp_zorder(x, y), grid_positions, np.array(list(grid_positions[1:]) + list(grid_positions[:1]))))
+
```

Comment:
Nice one liner for computing the grid indices, this part of the problem took me a long time to figure out. Very clean and easy to follow.